docs/articles/persistence/custom-persistence-provider.md

```diff
@@ -0,0 +1,467 @@
+        # Supported values:
+        # `repair-by-discard-old` : discard events from old writers,
+        #                           warning is logged
+        # `fail` : fail the replay, error is logged
+        # `warn` : log warning but emit events untouche
+        # `off` : disable this feature completely
+        mode = repair-by-discard-old
+
+        # It uses a look ahead buffer for analyzing the events.
+        # This defines the size (in number of events) of the buffer.
+        window-size = 100
+
+        # How many old writerUuid to remember
+        max-old-writers = 10
+
+        # Set this to `on` to enable detailed debug logging of each
+        # replayed event.
+        debug = off
+    }
+}
+```
+
+#### Pre-Start Requirement
+
+It is a good practice to allow user of your custom plugin to be able to auto initialize their back end event source from a blank slate. In order to do that, it is good practice to support `auto-initialize` setting in your journal HOCON settings.
+
+In order to support this, we will have to be able to stash all incoming messages while we're creating the tables in the event source. Here's an implementation that we use in our example code:  
+
+[!code-csharp[Startup](../../../src/examples/Akka.Persistence.Custom/Journal/SqliteJournal.cs?name=Startup "Journal startup sequence")]
+
+## Implementing Akka.Persistence SnapshotStore
+
+### Required Method Implementations
+
+#### LoadAsync
+
+```c#
+Task<SelectedSnapshot> LoadAsync(string persistenceId, SnapshotSelectionCriteria criteria)
+```
+
+* **`persistenceId`**: Identification of the persistent actor
+* **`criteria`**: Selection criteria for event loading
+
+This call is protected with a circuit-breaker.
+
+Asynchronously loads a snapshot.
+
+##### Code Sample
+
+`SelectSnapshotSql` in this example refers to this SQL query statement:
+
+[!code-csharp[SelectSnapshotSql](../../../src/examples/Akka.Persistence.Custom/Snapshot/SqliteSnapshotStore.cs?name=SelectSnapshotSql "SelectSnapshotSql SQL Statement")]
+
+SQLite code example:
+
+[!code-csharp[LoadAsync](../../../src/examples/Akka.Persistence.Custom/Snapshot/SqliteSnapshotStore.cs?name=LoadAsync "LoadAsync method implementation")]
+
+#### SaveAsync
+
+```c#
+Task SaveAsync(SnapshotMetadata metadata, object snapshot)
+```
+
+* **`metadata`**: Snapshot metadata
+* **`snapshot`**: The snapshot object instance to be stored
+
+This call is protected with a circuit-breaker
+
+Asynchronously saves a snapshot.
+
+##### Code Sample
+
+`InsertSnapshotSql` in this example refers to this SQL query statement:
+
+[!code-csharp[InsertSnapshotSql](../../../src/examples/Akka.Persistence.Custom/Snapshot/SqliteSnapshotStore.cs?name=InsertSnapshotSql "InsertSnapshotSql SQL Statement")]
+
+SQLite code example:
+
+[!code-csharp[SaveAsync](../../../src/examples/Akka.Persistence.Custom/Snapshot/SqliteSnapshotStore.cs?name=SaveAsync "SaveAsync method implementation")]
+
+#### DeleteAsync
+
+```c#
```

Comment:
Do we leave this as it is or make use of the range selection though the line number may change in the future?

Reply:
see above